apps/frontend/src/hooks/useWorkoutState.ts

```diff
@@ -0,0 +1,111 @@
+import { useLiveQuery } from 'dexie-react-hooks';
+import {
+  db,
+  defaultWorkoutState,
+  initWorkoutstate,
+  startNewWorkout,
+} from '../db';
+import { useEffect, useState } from 'react';
+import { millisToHoursMinutesAndSeconds } from '@dundring/utils';
+
+let hasPreCalculated = false;
+export const numberOfGraphDataPoints = 500;
+export const useWorkoutState = () => {
+  const [showRecoverPrompt, setShowRecoverPrompt] = useState(true);
+
+  const state =
+    useLiveQuery(() => db.workoutState.limit(1).last()) ?? defaultWorkoutState;
+
+  const trackedData =
+    useLiveQuery(
+      () =>
+        db.workoutDataPoint
+          .where('workoutNumber')
+          .equals(state.workoutNumber)
+          .reverse()
+          .toArray(),
+      [state.workoutNumber]
+    )
+      ?.toReversed()
+      ?.filter((dataPoint) => dataPoint.tracking) ?? [];
+
+  const lapData =
+    useLiveQuery(
+      () =>
+        db.workoutDataPoint
+          .where({
+            workoutNumber: state.workoutNumber,
+            lapNumber: state.lapNumber,
+          })
+          .reverse()
+          .toArray(),
+      [state.workoutNumber, state.lapNumber]
+    )
+      ?.toReversed()
+      ?.filter((dataPoint) => dataPoint.tracking) ?? [];
```

Comment:
could this filter be inside the where?

Reply:
Tried – when I did
```js
db.workoutDataPoint
  .where({
    workoutNumber: state.workoutNumber,
    lapNumber: state.lapNumber,
    tracking: true
})

```

I got a console warning suggesting to add an index for the three fields, but that resulted in an error. It appears to be impossible to add an index for a boolean, and some other people on the internet seem to have stumbled upon the same realization 😅 

I at least removed the extra reversing and moved the filter to be inside the the query